### PR TITLE
Refactors the procedural generation of lavaland and turf/closed/mineral

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 code: Lavaland is no longer generated in broken diagonals, instead utilizing  a map_generator datum.
 code: Bunch of unused turf paths removed
/:cl:

This replaces lavaland's old diagonal tunnel gen which was really horrendously jammed into asteroid floor code (?????) with Cellular Automata which runs in rust (PR for that here: https://github.com/tgstation/rust-g/pull/57 ). The new code is a bit cleaner, but also looks better.

VID: https://streamable.com/a45ke2

Things to do:
- Make an icemoon version
- Fix the roundstart atmos adjacency issues

I asked AnturK if this was an acceptable PR for this month; he said it was okay as long as I didn't add new areas, which I don't plan to do. But if anyone thinks this PR breaks the spirit of the month I'll open it again in december.

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]
rust-g will need to be built from a release tag, which should be fine after that PR is merged. Also dependencies.sh will need updating

**Comment by St0rmC4st3r** [comment]
Looks really great except for the badness in the file changes that should not be changed at all. Also pies? What procedural generation has to do with pies?

**Comment by Qustinnus** [comment]
is an issue with my master branch, it'll be fixed dw

**Comment by LemonInTheDark** [comment]
Ah it lives, ready for review I'd assume?

**Comment by Qustinnus** [comment]
Yeah, just don't merge this as it relies on https://github.com/tgstation/rust-g/pull/57

**Review by LemonInTheDark — COMMENTED** [review]
Made a start, I'll get more into it later TM

File: code/__DEFINES/rust_g.dm
Comment (Qustinnus) [reply code/__DEFINES/rust_g.dm:52]: the coords thing is no longer used to be fair, but in the future people might want to get a specific coord.

Comment (Qustinnus) [reply code/__DEFINES/rust_g.dm:52]: actually i can just make a define for that

File: code/__DEFINES/flags.dm
Comment (Qustinnus) [reply code/__DEFINES/flags.dm:74]: yes

**Comment by Qustinnus** [comment]
this wont pass checks until I get a release update of rust-g

**Comment by SpaceManiac** [comment]
is it just me or does this produce significantly fewer/smaller caves than before? is that the point? do we want that?

**Comment by SpaceManiac** [comment]
definitely an accident, it looks nothing like your vid, @Qustinnus I fixed a bunch of obvious math errors in the Rust code, maybe the caves only looked good because of them or something? can you confirm?

**Comment by Qustinnus** [comment]
I'll have another pass, you did invert one variable which I inverted in some commits, I tried playing with it to make it the same but I might have to tweak it some more.

**Comment by SpaceManiac** [comment]
That looks a lot better. PR content has my approval, I will handle cutting a rust-g release and merging.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**
